map_redcap_to_flywheel_service/utils/RCfield_classes.py

```diff
@@ -0,0 +1,436 @@
+
+        One example of why you wouldn't bother with an export
+        would be if the RECORD I'm looking at is for a repeated form measurement.
+
+        It will still have fields from all the NON repeated forms, but they'll
+        just be blank.  Obviously we don't want to overwrite anything with blank
+        data, so just don't export.
+
+        It's complicated, trust me.
+
+        """
+
+        repeat_form_key = "redcap_repeat_instrument"
+        repeat_num_key = "redcap_repeat_instance"
+        should_include = False
+        rnum = ''
+
+        # Case one, the record doesn't even have the repeat key in it, so there are not repeating forms.
+        # Include all fields in this case
+        if repeat_form_key not in record:
+            should_include = True
+            return should_include, rnum
+
+        # Case two, the record has the form, but it is blank...
+        if record.get(repeat_form_key) == '':
+            # 2a AND this is not a repeating field, include it because this means it's the record
+            # containing all NON repeating forms
+            if not self.repeating:
+                # No need to consider rnum since it's a non repeating field.
+                should_include = True
+                return should_include, rnum
+
+            # 2b otherwise if it IS a repeating field, then don't include it because it will be blank here
+            return should_include, rnum
+
+        # Case 3, the field IS from the repeating form that this record is showing
+        if self.form_name == record.get(repeat_form_key):
+            # Get the record number and include it
+            rnum = record.get(repeat_num_key)
+            should_include = True
+            return should_include, rnum
+
+        # Case 4, this record IS for a repeating form, but not THIS fields repeating form
+        # ALso catch all for all other cases.
+        return should_include, rnum
+
+
+    @staticmethod
+    def get_meta_object(foi, metadata, field_names):
+        meta = [m for m in metadata if m.get("field_name") == foi]
+
+        if not meta:
+            field = [f for f in field_names if f.get("original_field_name") == foi]
+            if not field:
+                log.error(f"no redcap field match for {foi}")
+                sys.exit(1)
+
+            field = field[0]
+
+            # If we're here it's some weird field that just indicates if the form is completed
+            form_name = foi
+            if foi.endswith("_complete"):
+                form_name = foi[: -len("_complete")]
+
+            meta = [
+                {
+                    "field_type": "dropdown",
+                    "field_name": field["original_field_name"],
+                    "field_label": field["export_field_name"],
+                    "form_name": form_name,
+                    "select_choices_or_calculations": "0, Incomplete | 1, unverified | 2, complete",
+                }
```

Comment:
does case mater in "select_choices_or_calculations" ?

Reply:
Good catch, though strictly speaking this doesn't matter, since their choice is stored as a number (0,1,2), this is just for a human translation of the number.  Changing it anyway, though.